src/zarr/storage/_utils.py

```diff
@@ -1,11 +1,61 @@
 from __future__ import annotations
 
+from pathlib import Path
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
     from zarr.core.buffer import Buffer
 
 
+def normalize_path(path: str | bytes | Path | None) -> str:
```

Comment:
Does this need to take `| None` still? Or have we replaced all those with `""` as the default?

Reply:
I have not changed any of the top-level defaults, so this still needs to handle `None`.